### PR TITLE
refresh global memory usage periodically

### PR DESCRIPTION
**Pull request by hicqu**
Signed-off-by: qupeng <qupeng@pingcap.com>

### What problem does this PR solve?

Refresh global memory usage periodically.

It introduces 2 functions `record_global_mem_usage` and `get_global_mem_usage`. The former is called periodically in the stat thread initialized in the server module, and the latter can be called anywhere to get the current memory usage.

BWT `SysQuota` is refactored to avoid lock the global `SYS_INFO` every times. 

### Check List 

Tests 

- No code

Side effects

### Release note 
* No release note

### CODE REVIEW

**Comment by rleungx** [comment]
_"it’s much better to keep the same instance of System around instead of recreating it multiple times."_

So I am not sure if it's the right way to recreate the system multiple times.

**Review by Connor1996 — COMMENTED** [review]
rest LGTM

File: components/tikv_util/src/sys/mod.rs [hunk 1]
```diff
@@ -6,108 +6,87 @@ pub mod cpu_time;
...
 
-        pub fn log_quota(&self) {
-            info!(
-                "memory limit in bytes: {}, cpu cores quota: {}",
-                self.memory_limit_in_bytes(),
-                self.cpu_cores_quota()
-            );
-        }
+    fn sysinfo_memory_limit_in_bytes() -> u64 {
+        let system = sysinfo::System::new_with_specifics(RefreshKind::new().with_memory());
```
Comment (Connor1996) [inline components/tikv_util/src/sys/mod.rs:75]: should it reuse the global sysinfo?

Comment (hicqu) [reply components/tikv_util/src/sys/mod.rs:75]: Refresh memory is very light and these functions are only called when validating `TiKvConfig` or environments. `SYS_INFO` is only for diagnoise now.
So both this and `record_global_memory_usage` doesn't reuse `SYS_INFO`.

**Review by Connor1996 — APPROVED** [review]

File: components/tikv_util/src/sys/mod.rs [hunk 2]
```diff
@@ -6,108 +6,87 @@ pub mod cpu_time;
...
-#[cfg(not(target_os = "linux"))]
-pub mod sys_quota {
-    use super::super::config::KIB;
-    use super::{SystemExt, SYS_INFO};
 
-    pub struct SysQuota {}
+    #[cfg(not(target_os = "linux"))]
+    pub fn cpu_cores_quota() -> f64 {
+        let cpu_num = num_cpus::get() as f64;
+        super::limit_cpu_cores_quota_by_env_var(cpu_num)
```
Comment (rleungx) [inline components/tikv_util/src/sys/mod.rs:47]: no need super?

**Pull request merged**